app/controllers/application_controller.rb

```diff
@@ -72,4 +72,11 @@ def encryptor
   def metrics_logger
     METRICS_LOGGER
   end
+
+  def ensure_visit_integrity
+    unless visit && visit.prisoner && visit.prisoner.prison_name.present?
+      redirect_to edit_prisoner_details_path,
+        notice: 'We’re missing some vital information'
```

Comment:
@leighmoney This is the message that they'll see if they go directly to the middle of the process, either by going to the URL directly or by doing something like clicking back after a session timeout. Previously, the app just blew up; now, they're sheep-dipped through the front gate again. It's not the name we need, as such, but *all* the information!